apps/mobile/src/components/register-adoption-form/adoption-form.component.test.tsx

```diff
@@ -0,0 +1,140 @@
+
+import RegisterAdoptionForm from './adoption-form.component';
+import { AdoptionSteps } from './adoption-form.types';
+import Routes from '../../routes';
+import Home from '../../screens/home/home.screen';
+import { renderWithProviders, screen, fireEvent, act, waitFor } from '../../test/test-utils';
+
+const mockShow = jest.fn();
+jest.mock('native-base', () => ({
+  ...jest.requireActual('native-base'),
+  useToast: () => ({
+    show: mockShow,
+    isActive: () => false,
+  }),
+}));
+
+const forwardStep = () => {
+  fireEvent.press(screen.getByText(/pr[oó]xima etapa/gi));
+};
+
+const fillOutTextInput = async (testId: string, value: string) => {
+  const input = await screen.findByTestId(`adoption-form-${testId}-input`);
+  fireEvent.changeText(input, value);
+};
+
+const pickOptionFromList = async (optionText: string | RegExp) => {
+  const option = await screen.findByText(optionText);
+  fireEvent.press(option);
+};
+
+const goToLastStep = async () => {
+  await fillOutTextInput('name', 'Bob');
+  forwardStep();
+  await fillOutTextInput('breed', 'pitbull');
+  forwardStep();
+  await pickOptionFromList(/c[aã]o/gi);
+  forwardStep();
+  await screen.findByTestId(`adoption-form-age-input`);
+  forwardStep();
+  await pickOptionFromList(/macho/i);
+  forwardStep();
+  await pickOptionFromList(/grande/i);
+  forwardStep();
+  await screen.findByText(/confirmar/i);
+};
+
+const Stack = createNativeStackNavigator();
+
+const MainNavigator = () => {
+  return (
+    <Stack.Navigator initialRouteName={Routes.RegisterAdoption}>
+      <Stack.Screen name={Routes.Home} component={Home} />
+      <Stack.Screen name={Routes.RegisterAdoption}>
+        {() => <RegisterAdoptionForm defaultValues={{ age: 2 }} />}
+      </Stack.Screen>
+    </Stack.Navigator>
+  );
+};
+
+const stepErrors: { step: AdoptionSteps; errorMessage: string }[] = [
+  {
+    step: AdoptionSteps.PetName,
+    errorMessage: `"name" is required`,
+  },
+  {
+    step: AdoptionSteps.PetBreed,
+    errorMessage: `"breed" is required`,
+  },
+  {
+    step: AdoptionSteps.PetType,
+    errorMessage: `"type" is required`,
+  },
+  {
+    step: AdoptionSteps.PetAge,
+    errorMessage: `"age" is required`,
+  },
+  {
+    step: AdoptionSteps.PetGender,
+    errorMessage: `"gender" is required`,
+  },
+  {
+    step: AdoptionSteps.PetSize,
+    errorMessage: `"size" is required`,
+  },
+];
+
+describe('AdoptionForm', () => {
+  describe('when the user presses the confirm button', () => {
+    describe('and the form state is valid', () => {
+      it('takes the user to the home screen', async () => {
+        renderWithProviders(<MainNavigator />);
+
+        await goToLastStep();
+
+        const confirmButton = screen.getByText(/confirmar/i);
+
+        await act(async () => {
+          fireEvent.press(confirmButton);
+          const home = await screen.findByText(/adoptions demo/i);
+          expect(home).toBeOnTheScreen();
+        });
```

Comment:
You don't really need to wrap your code under `acts` because RNTL already does that for you

See:
https://callstack.github.io/react-native-testing-library/docs/understanding-act#when-to-use-act

> When to use act
Thankfully, for these basic cases RNTL has got you covered as our render, update and fireEvent methods already wrap their calls in sync act so that you do not have to do it explicitly.